### ISSUE TITLE
jinja2-created global variable not processed correctly with "when" statement for include tasks

### ISSUE THREAD

**Issue by draxxris** [ansible/ansible#28670]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

base

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/ansible']
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### OS / ENVIRONMENT

Ubuntu 16.04 & Centos 7

##### SUMMARY

If you configure a global variable in your playbook derived from Jinja2 code, the variable is properly processed when calling it via "debug" task. However, when you use it as a "when" clause in an "include" task, ansible prints out warnings with the *full* un-processed Jinja2 code.

##### STEPS TO REPRODUCE

Please see playbook: https://github.com/polski-g/ansible-bug-report/tree/master

##### EXPECTED RESULTS

include tasks' "when" clause should use the post-processed variable rather than the raw jinja2 code.

##### ACTUAL RESULTS

You can change the `default(true)` to `default(false)` and you will see the issue in both cases.

Below, you can see that the role1:debug task works. But the role1:include(included.yml):debug complains about the variable, *but still prints out the proper value in the task*. Thus I think the issue is with the when: statement in the include stanza in tasks/main.yml

```
PLAY [localhost] ****************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [role1 : debug] ************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "do_something": "false"
}

TASK [role1 : debug] ************************************************************************************************************************************************************************************************************************
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {% if has_something|default(false) %}true{% else %}false{% endif %}

skipping: [localhost]

TASK [role1 : fail] *************************************************************************************************************************************************************************************************************************
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: {% if has_something|default(false) %}true{% else %}false{% endif %}

skipping: [localhost]
```

**Comment by sivel** [comment]
This should be resolved in the upcoming 2.4 release as part of https://github.com/ansible/ansible/pull/25092

Please verify that the `devel` code works as expected.

needs_info

**Comment by draxxris** [comment]
Yes, confirmed fixed via work done in #25092
